arroyo-api/src/pipelines.rs

```diff
@@ -303,6 +305,24 @@ impl TryInto<PipelineDef> for DbPipeline {
     }
 }
 
+pub(crate) async fn start_or_preview<'a>(
```

Comment:
TODO: Can this function be named something like create_or_preview?  "start" implies what we call an "update" which is just adjusting the state of the job.  Here though, start means to create a new pipeline.